luxjs/tests/forms/form.js

```diff
@@ -0,0 +1,185 @@
+define(function(require) {
+
+    describe("Test lux.form module", function() {
+
+        var $lux;
+        var successMessageSpy;
+        var errorMessageSpy;
+        var scope;
+
+        var testFormUtils = {
+            createForm: function () {
+                var form = {
+                    field: {
+                        type: 'form'
+                    },
+                    children: []
+                };
+                lux.forEach(arguments, function (attrs) {
+                    form['children'].push({field: attrs});
+                });
+                return form;
+            },
+            digest: function ($compile, $rootScope, template) {
+                scope = $rootScope.$new();
+                var element = $compile(template)(scope);
+                scope.$digest();
+                return element;
+            }
+        };
+
+        lux.formTests = {};
+
+        beforeEach(function () {
+            apiMock = createLuxApiMock();
+            var $luxMock = createLuxMock(apiMock);
+
+            angular.mock.module('lux.form', function($provide) {
+                $provide.value('$lux', $luxMock);
+            });
+
+            inject(function (_$lux_) {
+                $lux = _$lux_;
+            });
+
+            successMessageSpy = jasmine.createSpy();
+            errorMessageSpy = jasmine.createSpy();
+
+            $lux.messages = {};
+            $lux.messages.success = successMessageSpy;
+            $lux.messages.error = errorMessageSpy;
+        });
+
+        afterEach(function () {
+        });
+
+        it("basic input tests", inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
+                type: 'email',
+                name: 'login',
+                required: true
+            });
+
+            var element = testFormUtils.digest($compile, $rootScope,
+                '<div><lux-form data-options="lux.formTests.simple"></lux-form></div>'),
+                form = element.children();
+            //
+            var tags = form.children().children();
+            expect(tags[0].tagName).toBe('LABEL');
+            expect(tags[1].tagName).toBe('INPUT');
+
+            expect(tags[1].required).toBe(true);
+            expect(tags[1].type).toBe('email');
+            expect(tags[1].name).toBe('login');
+        }));
+
+        it("on success, email validate", inject(function($compile, $rootScope) {
+            lux.formTests.simple = testFormUtils.createForm({
```

Comment:
use a different attribute than ``simple``, don't reuse the same attribute as the previous test. ``lux`` is a global variable.